projects/igniteui-angular/src/lib/grids/headers/sorting-index.pipe.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { ISortingExpression } from '../../data-operations/sorting-expression.interface';
+import { IgxGridHeaderComponent } from './grid-header.component';
+
+@Pipe({
+  name: 'sortingIndex',
+  pure: true
+})
+export class SortingIndexPipe implements PipeTransform {
+
+  constructor(private igxHeader: IgxGridHeaderComponent) {}
+  transform(columnField: string, sortingExpressions: ISortingExpression[]): number {
+    const grid = this.igxHeader.grid;
+    let sortIndex = sortingExpressions.filter(expression => grid.getColumnByName(expression.fieldName) ?? false)
+                                      .findIndex(expression => expression.fieldName === columnField);
```

Comment:
```suggestion
    let sortIndex = sortingExpressions.findIndex(expression => expression.fieldName === columnField);
```

Reply:
@rkaraivanov I added the filter function in order to clear out the expressions, which have wrong field names, otherwise the sorting indicators will have wrong indexes

Reply:
@HristoP96 
Our API calls generate the correct sorting expression objects. The only way the column field would be different in the expression is if the user misspells the field name in which case I don't think we should be a spell checker 😃